### DISCUSSION TITLE
File transfer fails.

### DISCUSSION THREAD

**Discussion by Papablaubaer** [arendst/Tasmota#21790] [Q&A]
In the last few days I have tried to configure Tasmota on ESP32 automatically. Under 16-bit ESP I solved this with DecodeConfig, but the possibilities with Berry go far beyond what is possible with Tasmota on ESP16.
I played around a bit with .autoconf, and apart from a few small problems everything worked so far.
Then I tried to clean up my ESP32 with:

import persist
persist.zero()

import path
path.format(true)

I deleted all the old files.
Then curl should transfer the new .autoconf. This failed with very few exceptions, transferring using the upload button from the Tasmota interface worked much more often but not always. I tried everything imaginable to get curl to transfer my .autoconf, the result was always "not enough storage space". My file is not even 1Kb in size and 320Kb are free according to tasmota.
I tried limiting the speed with curl --limit-rate 1K and other values, but it always failed except for the first attempt.

Restarts between cleaning and transferring the data, with and without a power reset, didn't help either. Built-in wait times between cleaning and transferring didn't help either.

I thought about whether Tasmota has a secret memory and changes the file names with each attempt. I tried the latest development version. Nothing I did led to a good result.

In the end I compiled a version of Tasmota with an FTP server. What can I say, with FTP transfer via curl everything works as expected.

**Comment by s-hadinger** [comment]
I suppose that when you mention ESP16, you probably mean ESP8266? ESP8266 is 32-bit as well and share the same cpu core as Esp32. 

Anyways, I seems to me that you need to do a first request to the web file api before being able to upload a file (like done with the web UI). I'm not sure why and can't test until next week

**Comment by Papablaubaer** [comment]
Wie auch immer ich meine natürlich Tasmota und Tasmota32. Der Upload von Files ist, aus welchem Grund auch immer, irgendwie problematisch. Ich habe in anderem Zusammenhang schonmal in Discussion #21725 darauf hingewiesen. Die dort angebotenen Lösungen haben leider zu keinem Ergebnis geführt. Tasmota verschwindet manchmal im Nirwana, trotz aktiviertem Watchdog und der neuesten Entwicklerversion. Aber per FTP scheint alles so zu funktionieren wie erwartet, das kann ja ein guter Hinweis sein für die Entwickler und  für die Leute die sich mit ähnlichen Problemen beschäfigen.

Anyway, I mean Tasmota and Tasmota32 of course. Uploading files is, for whatever reason, somehow problematic. I have already pointed this out in another context in Discussion #21725. Unfortunately, the solutions offered there have not led to any results. Tasmota sometimes disappears into oblivion, despite the watchdog being activated and the latest developer version. But everything seems to work as expected via FTP, which could be a good indication for the developers and for people who are dealing with similar problems.

**Comment by Papablaubaer** [comment]
Ich habe gerade mal wieder ein bisschen mit Berry gemacht. 
Ich habe noch die alten curl befehle genutzt um meine Dateien hochzuladen. Plötzlich war mein Device wieder einmal im Nirvana verschwunden. Es scheint aber auch so zu sein das es ein Problem mit der Wifi Verbindung gibt. Ein "Reset 3" und Spannung weg hat geholfen, ich hab das früher bei meinen Versuchen mit Berry auch schon bemerkt, aber nicht in Zusammenhang gebracht. Kann es sein das da irgendwas in den falschen Speicher geschrieben wird ?  

I've just been doing a bit with Berry again.
I was still using the old curl commands to upload my files. Suddenly my device disappeared into oblivion again. But it also seems that there is a problem with the WiFi connection. A "Reset 3" and turning off the power helped, I noticed this earlier when I tried using Berry, but didn't connect it. Could it be that something is being written to the wrong memory?
